modules/widgets/blog-stats.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Blog Stats Widget.
+ *
+ * @since 4.5.0
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Disable direct access/execution to/of the widget code.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Blog Stats Widget.
+ *
+ * Displays all time stats for that site.
+ *
+ * @since 4.5.0
+ */
+class Jetpack_Blog_Stats_Widget extends WP_Widget {
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		$widget_ops = array(
+			'classname' => 'blog-stats',
+			'description' => __( 'Show a hit counter for your blog.', 'jetpack' ),
+			'customize_selective_refresh' => true,
+		);
+		parent::__construct(
+			'blog-stats',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Blog Stats', 'jetpack' ) ),
+			$widget_ops
+		);
+		$this->alt_option_name = 'widget_statscounter';
+	}
+
+	/**
+	 * Return an associative array of default values
+	 *
+	 * These values are used in new widgets.
+	 *
+	 * @return array Array of default values for the Widget's options
+	 */
+	public function defaults() {
+		return array(
+			'title' => __( 'Blog Stats', 'jetpack' ),
+			/* Translators: Number of views, plural */
+			'hits'  => __( 'hits', 'jetpack' ),
+		);
+	}
+
+	/**
+	 * Return All Time Stats for that blog.
+	 *
+	 * We query the WordPress.com Stats REST API endpoint.
+	 *
+	 * @uses stats_get_from_restapi(). That function caches data locally for 5 minutes.
+	 *
+	 * @return string|false $views All Time Stats for that blog.
+	 */
+	public function get_stats() {
+		// Get data from the WordPress.com Stats REST API endpoint.
+		$stats = stats_get_from_restapi( array( 'fields' => 'stats' ) );
+
+		if ( isset( $stats->stats->views ) ) {
+			return $stats->stats->views;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 *
+	 * @return void
+	 */
+	function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults() );
+		?>
+
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"><?php esc_html_e( 'Title:', 'jetpack' ); ?></label>
+			<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+		</p>
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'hits' ) ); ?>"><?php echo number_format_i18n( '12345' ); ?></label>
```

Comment:
I'm not sure this label makes sense. When I first looked at it, I asked myself, "Do I have 12,345 hits? That doesn't sound right."

In a way, I suppose it makes sense since it's the word we show after the number. So, we could change it to `page views` for instance.

Perhaps we could use a label like, "What noun should we use to describe each page view?"

And then put the number to left of the input?

<img width="443" alt="screen shot 2016-12-07 at 12 47 26 pm" src="https://cloud.githubusercontent.com/assets/1126811/20981729/61cd5c2c-bc7b-11e6-9ee4-8c8e688ed5ca.png">